vcd/catalogitem.go

```diff
@@ -46,7 +46,16 @@ func deleteCatalogItem(d *schema.ResourceData, vcdClient *VCDClient) error {
 	return nil
 }
 
-// Finds catalog item which can be vapp template OVA or media ISO file
+// vappTemplateToCatalogItem returns the catalog item corresponding to the associated vApp template
+func vappTemplateToCatalogItem(vappTemplateName string, catalog *govcd.Catalog) (*govcd.CatalogItem, error) {
+	item, err := catalog.GetCatalogItemByName(vappTemplateName, false)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving catalog item from vapp template %s: %s", vappTemplateName, err)
+	}
+	return item, nil
+}
+
+// Finds catalog item which can be vApp template OVA or media ISO file
 func findCatalogItem(d *schema.ResourceData, vcdClient *VCDClient, origin string) (*govcd.CatalogItem, error) {
```

Comment:
we should rename it `findVappTemplate`
* we filter only by template
* we use only in vapp template resource, aka now called catalog item resource

Reply:
We need a discussion on this. The whole catalogItem is a hybrid that should be refactored completely. I have tried to keep the fiction that we are dealing with catalogItem and not vApp template.

To make things right, we should implement a vApp template resource and deprecate catalog item. When I started with this PR, we agreed that this is not the time for such operation.

Reply:
ok. Then let's at least add at least TODO with explanation

Reply:
I will add a reference to issue #502